### PR TITLE
Fix issues with npm --prefix

### PR DESCRIPTION
**Pull request by atuchin-m**
Resolves https://github.com/brave/brave-browser/issues/19646
This fixes creating `acorn.cmd, acorn.ps1` in `src/brave` sometimes.

For Windows `npm --prefix` works in a strange way: it changes `npm bin` to the same as prefix which leads to the linked bug.
Therefore `npm run sync` and `npm --prefix=. run sync ` are not the same command.
To resolve this better to always use `cd` + `npm run` without prefix.

Linked PR in brave-core: https://github.com/brave/brave-core/pull/11212

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A
  - [ ] https://github.com/brave/brave-browser/wiki/Brave-Wallet

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -209,27 +209,27 @@
...
-    "gclient": "npm --prefix src/brave run gclient --",
-    "gn_check": "npm --prefix src/brave run gn_check --",
+    "apply_patches": "cd src/brave && npm run apply_patches --",
+    "audit_deps": "cd src/brave && npm run audit_deps --",
+    "build": "cd src/brave && npm run build --",
+    "chromium_rebase_l10n": "cd src/brave && npm run chromium_rebase_l10n --",
+    "cibuild": "cd src/brave && npm run cibuild --",
+    "create_dist": "cd src/brave && npm run create_dist --",
+    "delete_string_translations": "cd src/brave && npm run delete_string_translations --",
+    "gclient": "cd src/brave && cd src/brave && npm run gclient --",
```
Comment (bridiver) [inline package.json:219]: you have `cd src/brave` twice here

Comment (atuchin-m) [reply package.json]: Thanks, fixed

**Comment by atuchin-m** [comment]
BTW, other projects do the same: https://github.com/langium/langium/pull/95/files

Problems on windows with prefix are known issues https://github.com/npm/cli/issues/1290
Maybe they were fixed in the latest npm release.

**Review by iefremov — APPROVED** [review]
I'm all for it, but will defer the decision to someone who is npm-proficient

**Review by bridiver — APPROVED** [review]

**Review by petemill — APPROVED** [review]
Verified it doesn't leave the working directory changed 👍

**Pull request merged**